WordPress/Classes/Services/PostService.m

```diff
@@ -312,18 +312,22 @@ - (void)uploadPost:(AbstractPost *)post
     }
 }
 
-- (void)autoSave:(AbstractPost *)post
-         success:(void (^)(AbstractPost *post, NSString *previewURL))success
-         failure:(void (^)(NSError * _Nullable error))failure
-{
-    NSManagedObjectID *postObjectID = post.objectID;
+#pragma mark - Autosave Related
 
-    NSError *defaultError =
-        [NSError errorWithDomain:PostServiceErrorDomain
-                            code:0
-                        userInfo:@{ NSLocalizedDescriptionKey : @"Previews are unavailable for this kind of post." }];
+typedef void (^AutosaveFailureBlock)(NSError *error);
+typedef void (^AutosaveSuccessBlock)(RemotePost *post, NSString *previewURL);
 
-    void (^setPostAsFailedAndCallFailureBlock)(NSError *error) = ^(NSError *error) {
+- (NSError *)defaultAutosaveError
+{
+    return [NSError errorWithDomain:PostServiceErrorDomain
+                               code:0
+                           userInfo:@{ NSLocalizedDescriptionKey : @"Previews are unavailable for this kind of post." }];
+}
+
+- (AutosaveFailureBlock)wrappedAutosaveFailureBlock:(AbstractPost *)post failure:(void (^)(NSError * _Nullable error))failure
```

Comment:
We could probably pass `postObjectID` instead so this is more consistent with the `wrappedAutosaveSuccessBlock`'s signature.

Reply:
I considered that. I opted to keep the type AbstractPost since that's what we ultimately need to obtain. NSManagedObjectID wouldn't be a safe type in this case, strictly speaking, and so seems less future proof.